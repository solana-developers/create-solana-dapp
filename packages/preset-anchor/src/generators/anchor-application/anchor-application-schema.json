{
  "$schema": "http://json-schema.org/schema",
  "$id": "AnchorApplicationSchema",
  "title": "anchor-application-schema",
  "description": "Generate an Anchor application",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the application",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "skipFormat": {
      "description": "Skip formatting files",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "template": {
      "type": "string",
      "enum": ["counter", "hello-world"],
      "description": "The template to use",
      "default": "counter"
    }
  },
  "required": ["name"]
}
