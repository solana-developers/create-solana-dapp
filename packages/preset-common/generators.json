{
  "generators": {
    "common-template": {
      "factory": "./src/generators/common-template/common-template-generator",
      "schema": "./src/generators/common-template/common-template-schema.json",
      "description": "common-template generator"
    }
  }
}
