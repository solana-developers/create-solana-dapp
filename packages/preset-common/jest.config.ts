/* eslint-disable */
export default {
  displayName: 'preset-common',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../coverage/packages/preset-common',
  testEnvironment: 'node',
}
