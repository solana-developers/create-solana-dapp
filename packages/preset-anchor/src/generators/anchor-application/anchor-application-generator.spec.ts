import { readProjectConfiguration, Tree } from '@nx/devkit'
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing'
import { getRecursiveFileContents } from '@solana-developers/preset-common'
import {
  AnchorApplicationTemplate,
  anchorApplicationNormalizeSchema,
  AnchorApplicationNormalizedSchema,
} from '../../utils'
import { anchorApplicationGenerator } from './anchor-application-generator'
import { AnchorApplicationSchema } from './anchor-application-schema'

describe('anchor-application generator', () => {
  let tree: Tree
  const rawOptions: AnchorApplicationSchema = { name: 'anchor-app' }
  const options: AnchorApplicationNormalizedSchema = anchorApplicationNormalizeSchema(rawOptions)

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    await anchorApplicationGenerator(tree, options)
    const config = readProjectConfiguration(tree, options.name)
    expect(config).toBeDefined()
  })

  it.each([['counter'], ['hello-world']])('should generate app with "%s" template', async (template) => {
    await anchorApplicationGenerator(tree, { ...options, template: template as AnchorApplicationTemplate })

    const config = readProjectConfiguration(tree, options.name)
    const contents = getRecursiveFileContents(tree, config.root)
    expect(contents).toMatchSnapshot()

    const rootTsConfig = tree.read('tsconfig.base.json').toString()
    expect(rootTsConfig).toMatchSnapshot()
  })
})
