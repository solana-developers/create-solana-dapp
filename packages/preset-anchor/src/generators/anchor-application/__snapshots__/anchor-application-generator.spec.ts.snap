// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anchor-application generator should generate app with "counter" template 1`] = `
{
  ".eslintrc.json": {
    "content": [
      "{",
      ""extends": ["../.eslintrc.json"],",
      ""ignorePatterns": ["!**/*"],",
      ""overrides": [",
      "{",
      ""files": ["*.ts", "*.tsx", "*.js", "*.jsx"],",
      ""rules": {}",
      "},",
      "{",
      ""files": ["*.ts", "*.tsx"],",
      ""rules": {}",
      "},",
      "{",
      ""files": ["*.js", "*.jsx"],",
      ""rules": {}",
      "},",
      "{",
      ""files": ["*.json"],",
      ""parser": "jsonc-eslint-parser",",
      ""rules": {",
      ""@nx/dependency-checks": [",
      ""error",",
      "{",
      ""ignoredFiles": ["{projectRoot}/rollup.config.{js,ts,mjs,mts}"]",
      "}",
      "]",
      "}",
      "}",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/.eslintrc.json",
  },
  ".swcrc": {
    "content": [
      "{",
      ""jsc": {",
      ""target": "es2017",",
      ""parser": {",
      ""syntax": "typescript",",
      ""decorators": true,",
      ""dynamicImport": true",
      "},",
      ""transform": {",
      ""decoratorMetadata": true,",
      ""legacyDecorator": true",
      "},",
      ""keepClassNames": true,",
      ""externalHelpers": true,",
      ""loose": true",
      "},",
      ""module": {",
      ""type": "es6"",
      "},",
      ""sourceMaps": true,",
      ""exclude": [",
      ""jest.config.ts",",
      "".*\\\\.spec.tsx?$",",
      "".*\\\\.test.tsx?$",",
      ""./src/jest-setup.ts$",",
      ""./**/jest-setup.ts$",",
      "".*.js$"",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/.swcrc",
  },
  "Anchor.toml": {
    "content": [
      "[features]",
      "seeds = false",
      "skip-lint = false",
      "[programs.localnet]",
      "counter = "EuJfgknXicf6iZASeagt5U6XDDPpLTQrvWPW31vgAwo8"",
      "[registry]",
      "url = "https://api.apr.dev"",
      "[provider]",
      "cluster = "Localnet"",
      "wallet = "~/.config/solana/id.json"",
      "[scripts]",
      "test = "../node_modules/.bin/nx run anchor-app:jest"",
      "[test.validator]",
      "bind_address = "127.0.0.1"",
    ],
    "isBinary": false,
    "path": "anchor-app/Anchor.toml",
  },
  "Cargo.lock": {
    "content": null,
    "isBinary": true,
    "path": "anchor-app/Cargo.lock",
  },
  "Cargo.toml": {
    "content": [
      "[workspace]",
      "members = [",
      ""programs/*"",
      "]",
      "resolver = "2"",
      "[profile.release]",
      "overflow-checks = true",
      "lto = "fat"",
      "codegen-units = 1",
      "[profile.release.build-override]",
      "opt-level = 3",
      "incremental = false",
      "codegen-units = 1",
    ],
    "isBinary": false,
    "path": "anchor-app/Cargo.toml",
  },
  "README.md": {
    "content": [
      "# anchor-app",
      "This library was generated with [Nx](https://nx.dev).",
      "## Building",
      "Run \`nx build anchor-app\` to build the library.",
      "## Running unit tests",
      "Run \`nx test anchor-app\` to execute the unit tests via [Jest](https://jestjs.io).",
    ],
    "isBinary": false,
    "path": "anchor-app/README.md",
  },
  "jest.config.ts": {
    "content": [
      "/* eslint-disable */",
      "import { readFileSync } from 'fs';",
      "// Reading the SWC compilation config and remove the "exclude"",
      "// for the test files to be compiled by SWC",
      "const { exclude: _, ...swcJestConfig } = JSON.parse(",
      "readFileSync(\`\${__dirname}/.swcrc\`, 'utf-8')",
      ");",
      "// disable .swcrc look-up by SWC core because we're passing in swcJestConfig ourselves.",
      "// If we do not disable this, SWC Core will read .swcrc and won't transform our test files due to "exclude"",
      "if (swcJestConfig.swcrc === undefined) {",
      "swcJestConfig.swcrc = false;",
      "}",
      "// Uncomment if using global setup/teardown files being transformed via swc",
      "// https://nx.dev/packages/jest/documents/overview#global-setup/teardown-with-nx-libraries",
      "// jest needs EsModule Interop to find the default exported setup/teardown functions",
      "// swcJestConfig.module.noInterop = false;",
      "export default {",
      "displayName: 'anchor-app',",
      "preset: '../jest.preset.js',",
      "transform: {",
      "'^.+\\\\.[tj]s$': ['@swc/jest', swcJestConfig],",
      "},",
      "moduleFileExtensions: ['ts', 'js', 'html'],",
      "testEnvironment: '',",
      "coverageDirectory: '../coverage/anchor-app',",
      "};",
    ],
    "isBinary": false,
    "path": "anchor-app/jest.config.ts",
  },
  "migrations": {
    "children": {
      "deploy.ts": {
        "content": [
          "// Migrations are an early feature. Currently, they're nothing more than this",
          "// single deploy script that's invoked from the CLI, injecting a provider",
          "// configured from the workspace's Anchor.toml.",
          "const anchor = require('@coral-xyz/anchor');",
          "module.exports = async function (provider) {",
          "// Configure client to use the provider.",
          "anchor.setProvider(provider);",
          "// Add your deploy script here.",
          "};",
        ],
        "isBinary": false,
        "path": "anchor-app/migrations/deploy.ts",
      },
    },
    "path": "anchor-app/migrations",
  },
  "package.json": {
    "content": [
      "{",
      ""name": "@proj/anchor-app",",
      ""version": "0.0.1",",
      ""dependencies": {},",
      ""type": "commonjs",",
      ""main": "./index.cjs",",
      ""module": "./index.js"",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/package.json",
  },
  "programs": {
    "children": {
      "counter": {
        "children": {
          "Cargo.toml": {
            "content": [
              "[package]",
              "name = "counter"",
              "version = "0.1.0"",
              "description = "Created with Anchor"",
              "edition = "2021"",
              "[lib]",
              "crate-type = ["cdylib", "lib"]",
              "name = "counter"",
              "[features]",
              "no-entrypoint = []",
              "no-idl = []",
              "no-log-ix-name = []",
              "cpi = ["no-entrypoint"]",
              "default = []",
              "[dependencies]",
              "anchor-lang = "0.28.0"",
            ],
            "isBinary": false,
            "path": "anchor-app/programs/counter/Cargo.toml",
          },
          "Xargo.toml": {
            "content": [
              "[target.bpfel-unknown-unknown.dependencies.std]",
              "features = []",
            ],
            "isBinary": false,
            "path": "anchor-app/programs/counter/Xargo.toml",
          },
          "src": {
            "children": {
              "lib.rs": {
                "content": [
                  "#![allow(clippy::result_large_err)]",
                  "use anchor_lang::prelude::*;",
                  "declare_id!("EuJfgknXicf6iZASeagt5U6XDDPpLTQrvWPW31vgAwo8");",
                  "#[program]",
                  "pub mod counter {",
                  "use super::*;",
                  "pub fn initialize_counter(_ctx: Context<InitializeCounter>) -> Result<()> {",
                  "Ok(())",
                  "}",
                  "pub fn increment(ctx: Context<Increment>) -> Result<()> {",
                  "ctx.accounts.counter.count = ctx.accounts.counter.count.checked_add(1).unwrap();",
                  "Ok(())",
                  "}",
                  "}",
                  "#[derive(Accounts)]",
                  "pub struct InitializeCounter<'info> {",
                  "#[account(mut)]",
                  "pub payer: Signer<'info>,",
                  "#[account(",
                  "init,",
                  "space = 8 + Counter::INIT_SPACE,",
                  "payer = payer",
                  ")]",
                  "pub counter: Account<'info, Counter>,",
                  "pub system_program: Program<'info, System>,",
                  "}",
                  "#[derive(Accounts)]",
                  "pub struct Increment<'info> {",
                  "#[account(mut)]",
                  "pub counter: Account<'info, Counter>,",
                  "}",
                  "#[account]",
                  "#[derive(InitSpace)]",
                  "pub struct Counter {",
                  "count: u64,",
                  "}",
                ],
                "isBinary": false,
                "path": "anchor-app/programs/counter/src/lib.rs",
              },
            },
            "path": "anchor-app/programs/counter/src",
          },
        },
        "path": "anchor-app/programs/counter",
      },
    },
    "path": "anchor-app/programs",
  },
  "project.json": {
    "content": [
      "{",
      ""name": "anchor-app",",
      ""$schema": "../node_modules/nx/schemas/project-schema.json",",
      ""sourceRoot": "anchor-app/src",",
      ""projectType": "library",",
      ""targets": {",
      ""build": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor build"],",
      ""parallel": false",
      "}",
      "},",
      ""lint": {",
      ""executor": "@nx/eslint:lint",",
      ""outputs": ["{options.outputFile}"],",
      ""options": {",
      ""lintFilePatterns": ["anchor-app/**/*.ts", "anchor-app/package.json"]",
      "}",
      "},",
      ""test": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor test"],",
      ""parallel": false",
      "}",
      "},",
      ""clean": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor clean"],",
      ""parallel": false",
      "}",
      "},",
      ""deploy": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor deploy"],",
      ""parallel": false",
      "}",
      "},",
      ""localnet": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor localnet"],",
      ""parallel": false",
      "}",
      "},",
      ""publish": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor publish"],",
      ""parallel": false",
      "}",
      "},",
      ""jest": {",
      ""executor": "@nx/jest:jest",",
      ""outputs": ["{workspaceRoot}/coverage/{projectRoot}"],",
      ""options": {",
      ""jestConfig": "anchor-app/jest.config.ts"",
      "}",
      "}",
      "},",
      ""tags": []",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/project.json",
  },
  "src": {
    "children": {
      "index.ts": {
        "content": [
          "import { Program } from '@coral-xyz/anchor';",
          "import { IDL as CounterIDL } from '../target/types/counter';",
          "import type { Counter } from '../target/types/counter';",
          "export { Counter, CounterIDL };",
          "export type CounterProgram = Program<Counter>;",
        ],
        "isBinary": false,
        "path": "anchor-app/src/index.ts",
      },
    },
    "path": "anchor-app/src",
  },
  "target": {
    "children": {
      "types": {
        "children": {
          "counter.ts": {
            "content": [
              "export type Counter = {",
              "version: '0.1.0';",
              "name: 'counter';",
              "instructions: [",
              "{",
              "name: 'initializeCounter';",
              "accounts: [",
              "{",
              "name: 'payer';",
              "isMut: true;",
              "isSigner: true;",
              "},",
              "{",
              "name: 'counter';",
              "isMut: true;",
              "isSigner: true;",
              "},",
              "{",
              "name: 'systemProgram';",
              "isMut: false;",
              "isSigner: false;",
              "}",
              "];",
              "args: [];",
              "},",
              "{",
              "name: 'increment';",
              "accounts: [",
              "{",
              "name: 'counter';",
              "isMut: true;",
              "isSigner: false;",
              "}",
              "];",
              "args: [];",
              "}",
              "];",
              "accounts: [",
              "{",
              "name: 'counter';",
              "type: {",
              "kind: 'struct';",
              "fields: [",
              "{",
              "name: 'count';",
              "type: 'u64';",
              "}",
              "];",
              "};",
              "}",
              "];",
              "};",
              "export const IDL: Counter = {",
              "version: '0.1.0',",
              "name: 'counter',",
              "instructions: [",
              "{",
              "name: 'initializeCounter',",
              "accounts: [",
              "{",
              "name: 'payer',",
              "isMut: true,",
              "isSigner: true,",
              "},",
              "{",
              "name: 'counter',",
              "isMut: true,",
              "isSigner: true,",
              "},",
              "{",
              "name: 'systemProgram',",
              "isMut: false,",
              "isSigner: false,",
              "},",
              "],",
              "args: [],",
              "},",
              "{",
              "name: 'increment',",
              "accounts: [",
              "{",
              "name: 'counter',",
              "isMut: true,",
              "isSigner: false,",
              "},",
              "],",
              "args: [],",
              "},",
              "],",
              "accounts: [",
              "{",
              "name: 'counter',",
              "type: {",
              "kind: 'struct',",
              "fields: [",
              "{",
              "name: 'count',",
              "type: 'u64',",
              "},",
              "],",
              "},",
              "},",
              "],",
              "};",
            ],
            "isBinary": false,
            "path": "anchor-app/target/types/counter.ts",
          },
        },
        "path": "anchor-app/target/types",
      },
    },
    "path": "anchor-app/target",
  },
  "tests": {
    "children": {
      "counter.spec.ts": {
        "content": [
          "import * as anchor from '@coral-xyz/anchor';",
          "import { Program } from '@coral-xyz/anchor';",
          "import { Keypair } from '@solana/web3.js';",
          "import { Counter } from '../target/types/counter';",
          "describe('counter', () => {",
          "// Configure the client to use the local cluster.",
          "const provider = anchor.AnchorProvider.env();",
          "anchor.setProvider(provider);",
          "const payer = provider.wallet as anchor.Wallet;",
          "const program = anchor.workspace.Counter as Program<Counter>;",
          "// Generate a new keypair for the counter account",
          "const counterKeypair = new Keypair();",
          "it('Initialize Counter', async () => {",
          "await program.methods",
          ".initializeCounter()",
          ".accounts({",
          "counter: counterKeypair.publicKey,",
          "payer: payer.publicKey,",
          "})",
          ".signers([counterKeypair])",
          ".rpc();",
          "const currentCount = await program.account.counter.fetch(",
          "counterKeypair.publicKey",
          ");",
          "expect(currentCount.count.toNumber()).toEqual(0);",
          "});",
          "it('Increment Counter', async () => {",
          "await program.methods",
          ".increment()",
          ".accounts({ counter: counterKeypair.publicKey })",
          ".rpc();",
          "const currentCount = await program.account.counter.fetch(",
          "counterKeypair.publicKey",
          ");",
          "expect(currentCount.count.toNumber()).toEqual(1);",
          "});",
          "it('Increment Counter Again', async () => {",
          "await program.methods",
          ".increment()",
          ".accounts({ counter: counterKeypair.publicKey })",
          ".rpc();",
          "const currentCount = await program.account.counter.fetch(",
          "counterKeypair.publicKey",
          ");",
          "expect(currentCount.count.toNumber()).toEqual(2);",
          "});",
          "});",
        ],
        "isBinary": false,
        "path": "anchor-app/tests/counter.spec.ts",
      },
    },
    "path": "anchor-app/tests",
  },
  "tsconfig.json": {
    "content": [
      "{",
      ""extends": "../tsconfig.base.json",",
      ""compilerOptions": {",
      ""module": "commonjs"",
      "},",
      ""files": [],",
      ""include": [],",
      ""references": [",
      "{",
      ""path": "./tsconfig.lib.json"",
      "},",
      "{",
      ""path": "./tsconfig.spec.json"",
      "}",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/tsconfig.json",
  },
  "tsconfig.lib.json": {
    "content": [
      "{",
      ""extends": "./tsconfig.json",",
      ""compilerOptions": {",
      ""outDir": "../dist/out-tsc",",
      ""declaration": true,",
      ""types": ["node"]",
      "},",
      ""include": ["src/**/*.ts"],",
      ""exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/tsconfig.lib.json",
  },
  "tsconfig.spec.json": {
    "content": [
      "{",
      ""extends": "./tsconfig.json",",
      ""compilerOptions": {",
      ""outDir": "../dist/out-tsc",",
      ""module": "commonjs",",
      ""types": ["jest", "node"]",
      "},",
      ""include": [",
      ""jest.config.ts",",
      ""src/**/*.test.ts",",
      ""src/**/*.spec.ts",",
      ""src/**/*.d.ts"",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/tsconfig.spec.json",
  },
}
`;

exports[`anchor-application generator should generate app with "counter" template 2`] = `
"{
  "compilerOptions": {
    "paths": {
      "@proj/anchor-app": ["anchor-app/src/index.ts"]
    }
  }
}
"
`;

exports[`anchor-application generator should generate app with "hello-world" template 1`] = `
{
  ".eslintrc.json": {
    "content": [
      "{",
      ""extends": ["../.eslintrc.json"],",
      ""ignorePatterns": ["!**/*"],",
      ""overrides": [",
      "{",
      ""files": ["*.ts", "*.tsx", "*.js", "*.jsx"],",
      ""rules": {}",
      "},",
      "{",
      ""files": ["*.ts", "*.tsx"],",
      ""rules": {}",
      "},",
      "{",
      ""files": ["*.js", "*.jsx"],",
      ""rules": {}",
      "},",
      "{",
      ""files": ["*.json"],",
      ""parser": "jsonc-eslint-parser",",
      ""rules": {",
      ""@nx/dependency-checks": [",
      ""error",",
      "{",
      ""ignoredFiles": ["{projectRoot}/rollup.config.{js,ts,mjs,mts}"]",
      "}",
      "]",
      "}",
      "}",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/.eslintrc.json",
  },
  ".swcrc": {
    "content": [
      "{",
      ""jsc": {",
      ""target": "es2017",",
      ""parser": {",
      ""syntax": "typescript",",
      ""decorators": true,",
      ""dynamicImport": true",
      "},",
      ""transform": {",
      ""decoratorMetadata": true,",
      ""legacyDecorator": true",
      "},",
      ""keepClassNames": true,",
      ""externalHelpers": true,",
      ""loose": true",
      "},",
      ""module": {",
      ""type": "es6"",
      "},",
      ""sourceMaps": true,",
      ""exclude": [",
      ""jest.config.ts",",
      "".*\\\\.spec.tsx?$",",
      "".*\\\\.test.tsx?$",",
      ""./src/jest-setup.ts$",",
      ""./**/jest-setup.ts$",",
      "".*.js$"",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/.swcrc",
  },
  "Anchor.toml": {
    "content": [
      "[features]",
      "seeds = false",
      "skip-lint = false",
      "[programs.localnet]",
      "hello_world = "EuJfgknXicf6iZASeagt5U6XDDPpLTQrvWPW31vgAwo8"",
      "[registry]",
      "url = "https://api.apr.dev"",
      "[provider]",
      "cluster = "Localnet"",
      "wallet = "~/.config/solana/id.json"",
      "[scripts]",
      "test = "../node_modules/.bin/nx run anchor-app:jest"",
      "[test.validator]",
      "bind_address = "127.0.0.1"",
    ],
    "isBinary": false,
    "path": "anchor-app/Anchor.toml",
  },
  "Cargo.lock": {
    "content": null,
    "isBinary": true,
    "path": "anchor-app/Cargo.lock",
  },
  "Cargo.toml": {
    "content": [
      "[workspace]",
      "members = [",
      ""programs/*"",
      "]",
      "resolver = "2"",
      "[profile.release]",
      "overflow-checks = true",
      "lto = "fat"",
      "codegen-units = 1",
      "[profile.release.build-override]",
      "opt-level = 3",
      "incremental = false",
      "codegen-units = 1",
    ],
    "isBinary": false,
    "path": "anchor-app/Cargo.toml",
  },
  "README.md": {
    "content": [
      "# anchor-app",
      "This library was generated with [Nx](https://nx.dev).",
      "## Building",
      "Run \`nx build anchor-app\` to build the library.",
      "## Running unit tests",
      "Run \`nx test anchor-app\` to execute the unit tests via [Jest](https://jestjs.io).",
    ],
    "isBinary": false,
    "path": "anchor-app/README.md",
  },
  "jest.config.ts": {
    "content": [
      "/* eslint-disable */",
      "import { readFileSync } from 'fs';",
      "// Reading the SWC compilation config and remove the "exclude"",
      "// for the test files to be compiled by SWC",
      "const { exclude: _, ...swcJestConfig } = JSON.parse(",
      "readFileSync(\`\${__dirname}/.swcrc\`, 'utf-8')",
      ");",
      "// disable .swcrc look-up by SWC core because we're passing in swcJestConfig ourselves.",
      "// If we do not disable this, SWC Core will read .swcrc and won't transform our test files due to "exclude"",
      "if (swcJestConfig.swcrc === undefined) {",
      "swcJestConfig.swcrc = false;",
      "}",
      "// Uncomment if using global setup/teardown files being transformed via swc",
      "// https://nx.dev/packages/jest/documents/overview#global-setup/teardown-with-nx-libraries",
      "// jest needs EsModule Interop to find the default exported setup/teardown functions",
      "// swcJestConfig.module.noInterop = false;",
      "export default {",
      "displayName: 'anchor-app',",
      "preset: '../jest.preset.js',",
      "transform: {",
      "'^.+\\\\.[tj]s$': ['@swc/jest', swcJestConfig],",
      "},",
      "moduleFileExtensions: ['ts', 'js', 'html'],",
      "testEnvironment: '',",
      "coverageDirectory: '../coverage/anchor-app',",
      "};",
    ],
    "isBinary": false,
    "path": "anchor-app/jest.config.ts",
  },
  "migrations": {
    "children": {
      "deploy.ts": {
        "content": [
          "// Migrations are an early feature. Currently, they're nothing more than this",
          "// single deploy script that's invoked from the CLI, injecting a provider",
          "// configured from the workspace's Anchor.toml.",
          "const anchor = require('@coral-xyz/anchor');",
          "module.exports = async function (provider) {",
          "// Configure client to use the provider.",
          "anchor.setProvider(provider);",
          "// Add your deploy script here.",
          "};",
        ],
        "isBinary": false,
        "path": "anchor-app/migrations/deploy.ts",
      },
    },
    "path": "anchor-app/migrations",
  },
  "package.json": {
    "content": [
      "{",
      ""name": "@proj/anchor-app",",
      ""version": "0.0.1",",
      ""dependencies": {},",
      ""type": "commonjs",",
      ""main": "./index.cjs",",
      ""module": "./index.js"",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/package.json",
  },
  "programs": {
    "children": {
      "hello-world": {
        "children": {
          "Cargo.toml": {
            "content": [
              "[package]",
              "name = "hello-world"",
              "version = "0.1.0"",
              "description = "Created with Anchor"",
              "edition = "2021"",
              "[lib]",
              "crate-type = ["cdylib", "lib"]",
              "name = "hello_world"",
              "[features]",
              "no-entrypoint = []",
              "no-idl = []",
              "no-log-ix-name = []",
              "cpi = ["no-entrypoint"]",
              "default = []",
              "[dependencies]",
              "anchor-lang = "0.28.0"",
            ],
            "isBinary": false,
            "path": "anchor-app/programs/hello-world/Cargo.toml",
          },
          "Xargo.toml": {
            "content": [
              "[target.bpfel-unknown-unknown.dependencies.std]",
              "features = []",
            ],
            "isBinary": false,
            "path": "anchor-app/programs/hello-world/Xargo.toml",
          },
          "src": {
            "children": {
              "lib.rs": {
                "content": [
                  "use anchor_lang::prelude::*;",
                  "declare_id!("EuJfgknXicf6iZASeagt5U6XDDPpLTQrvWPW31vgAwo8");",
                  "#[program]",
                  "pub mod hello_world {",
                  "use super::*;",
                  "pub fn initialize(_ctx: Context<Initialize>) -> Result<()> {",
                  "msg!("Hello World!");",
                  "Ok(())",
                  "}",
                  "}",
                  "#[derive(Accounts)]",
                  "pub struct Initialize {}",
                ],
                "isBinary": false,
                "path": "anchor-app/programs/hello-world/src/lib.rs",
              },
            },
            "path": "anchor-app/programs/hello-world/src",
          },
        },
        "path": "anchor-app/programs/hello-world",
      },
    },
    "path": "anchor-app/programs",
  },
  "project.json": {
    "content": [
      "{",
      ""name": "anchor-app",",
      ""$schema": "../node_modules/nx/schemas/project-schema.json",",
      ""sourceRoot": "anchor-app/src",",
      ""projectType": "library",",
      ""targets": {",
      ""build": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor build"],",
      ""parallel": false",
      "}",
      "},",
      ""lint": {",
      ""executor": "@nx/eslint:lint",",
      ""outputs": ["{options.outputFile}"],",
      ""options": {",
      ""lintFilePatterns": ["anchor-app/**/*.ts", "anchor-app/package.json"]",
      "}",
      "},",
      ""test": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor test"],",
      ""parallel": false",
      "}",
      "},",
      ""clean": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor clean"],",
      ""parallel": false",
      "}",
      "},",
      ""deploy": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor deploy"],",
      ""parallel": false",
      "}",
      "},",
      ""localnet": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor localnet"],",
      ""parallel": false",
      "}",
      "},",
      ""publish": {",
      ""executor": "nx:run-commands",",
      ""options": {",
      ""cwd": "anchor-app",",
      ""commands": ["anchor publish"],",
      ""parallel": false",
      "}",
      "},",
      ""jest": {",
      ""executor": "@nx/jest:jest",",
      ""outputs": ["{workspaceRoot}/coverage/{projectRoot}"],",
      ""options": {",
      ""jestConfig": "anchor-app/jest.config.ts"",
      "}",
      "}",
      "},",
      ""tags": []",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/project.json",
  },
  "src": {
    "children": {
      "index.ts": {
        "content": [
          "import { Program } from '@coral-xyz/anchor';",
          "import { IDL as HelloWorldIDL } from '../target/types/hello_world';",
          "import type { HelloWorld } from '../target/types/hello_world';",
          "export { HelloWorld, HelloWorldIDL };",
          "export type HelloWorldProgram = Program<HelloWorld>;",
        ],
        "isBinary": false,
        "path": "anchor-app/src/index.ts",
      },
    },
    "path": "anchor-app/src",
  },
  "target": {
    "children": {
      "types": {
        "children": {
          "hello_world.ts": {
            "content": [
              "export type HelloWorld = {",
              "version: '0.1.0';",
              "name: 'hello_world';",
              "instructions: [",
              "{",
              "name: 'initialize';",
              "accounts: [];",
              "args: [];",
              "}",
              "];",
              "};",
              "export const IDL: HelloWorld = {",
              "version: '0.1.0',",
              "name: 'hello_world',",
              "instructions: [",
              "{",
              "name: 'initialize',",
              "accounts: [],",
              "args: [],",
              "},",
              "],",
              "};",
            ],
            "isBinary": false,
            "path": "anchor-app/target/types/hello_world.ts",
          },
        },
        "path": "anchor-app/target/types",
      },
    },
    "path": "anchor-app/target",
  },
  "tests": {
    "children": {
      "hello-world.spec.ts": {
        "content": [
          "import * as anchor from '@coral-xyz/anchor';",
          "import { Program } from '@coral-xyz/anchor';",
          "import { HelloWorld } from '../target/types/hello_world';",
          "describe('hello-world', () => {",
          "// Configure the client to use the local cluster.",
          "anchor.setProvider(anchor.AnchorProvider.env());",
          "const program = anchor.workspace.HelloWorld as Program<HelloWorld>;",
          "it('Is initialized!', async () => {",
          "// Add your test here.",
          "const tx = await program.methods.initialize().rpc();",
          "console.log('Your transaction signature', tx);",
          "});",
          "});",
        ],
        "isBinary": false,
        "path": "anchor-app/tests/hello-world.spec.ts",
      },
    },
    "path": "anchor-app/tests",
  },
  "tsconfig.json": {
    "content": [
      "{",
      ""extends": "../tsconfig.base.json",",
      ""compilerOptions": {",
      ""module": "commonjs"",
      "},",
      ""files": [],",
      ""include": [],",
      ""references": [",
      "{",
      ""path": "./tsconfig.lib.json"",
      "},",
      "{",
      ""path": "./tsconfig.spec.json"",
      "}",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/tsconfig.json",
  },
  "tsconfig.lib.json": {
    "content": [
      "{",
      ""extends": "./tsconfig.json",",
      ""compilerOptions": {",
      ""outDir": "../dist/out-tsc",",
      ""declaration": true,",
      ""types": ["node"]",
      "},",
      ""include": ["src/**/*.ts"],",
      ""exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/tsconfig.lib.json",
  },
  "tsconfig.spec.json": {
    "content": [
      "{",
      ""extends": "./tsconfig.json",",
      ""compilerOptions": {",
      ""outDir": "../dist/out-tsc",",
      ""module": "commonjs",",
      ""types": ["jest", "node"]",
      "},",
      ""include": [",
      ""jest.config.ts",",
      ""src/**/*.test.ts",",
      ""src/**/*.spec.ts",",
      ""src/**/*.d.ts"",
      "]",
      "}",
    ],
    "isBinary": false,
    "path": "anchor-app/tsconfig.spec.json",
  },
}
`;

exports[`anchor-application generator should generate app with "hello-world" template 2`] = `
"{
  "compilerOptions": {
    "paths": {
      "@proj/anchor-app": ["anchor-app/src/index.ts"]
    }
  }
}
"
`;
